### PR TITLE
Added 7 New Filters

### PR DESCRIPTION
**Pull request by madhavrathi**
Fixes issue #103 

Changes: Added 7 New Filters
1) Blue Alpha
2)Pink Alpha
3)Polaroid
4)Blur
5)Sharp
6)Edge
7)Fuzz

Screenshots for the change: 
screenshot_2017-03-12-17-47-01-775_vn mbm phimp me
screenshot_2017-03-12-17-46-52-762_vn mbm phimp me
screenshot_2017-03-12-17-46-45-583_vn mbm phimp me

### CODE REVIEW

**Comment by CloudyPadmal** [comment]
Looks awesome! :D Great work!! :100:

File: app/src/main/java/vn/mbm/phimp/me/Camera.java [hunk 1]
```diff
@@ -56,7 +58,11 @@ public void onClick(View v) {
 		String filename = "phimp.me_" + d.getTime() + ".jpg";
         _path = Environment.getExternalStorageDirectory() + "/phimp.me/take_photo/"+filename;
         Log.i("Danh","Path 1 :"+_path);
+
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera.java]: Please avoid blank lines which are not needed.

File: app/src/main/java/vn/mbm/phimp/me/Camera.java [hunk 2]
```diff
@@ -56,7 +58,11 @@ public void onClick(View v) {
...
 	}
+
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera.java]: Same here, delete these blank lines

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 3]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
                 });
 
         findViewById(R.id.btnSaphia).setOnClickListener(
+				new View.OnClickListener() {
+				    public void onClick(View v)
+				    {
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Make this single line like:
`public void onClick(View v) {`

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 4]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
+				    	try{
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Again a blank line

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 5]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
+				    		modifiedBitmap = null;
+							modifiedBitmap = convertToSepia(flippedImaged);
+							mImageView.setImageBitmap(changeBrightness(
+									modifiedBitmap, brightnessValue));
+							mBitmap = modifiedBitmap;
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Blank line

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 6]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
                 });
 
         findViewById(R.id.btnSaphia).setOnClickListener(
+				new View.OnClickListener() {
+				    public void onClick(View v)
+				    {
+				    	try{
+
+				    		modifiedBitmap = null;
+							modifiedBitmap = convertToSepia(flippedImaged);
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Intend the line correctly

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 7]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
+							mImageView.setImageBitmap(changeBrightness(
+									modifiedBitmap, brightnessValue));
+							mBitmap = modifiedBitmap;
+
+				    	}catch(OutOfMemoryError o){
+				    		mBitmapResize=getResizedBitmap(mBitmap, (mBitmap.getHeight()/4), (mBitmap.getWidth()/4));
+				    		//mBitmapResize=getResizedBitmap(p[0], (mBitmap.getHeight()/2), (mBitmap.getWidth()/2));
+							modifiedBitmap=flippedImaged=mBitmapResize;
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Intend this.

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 8]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
+
+				    		modifiedBitmap = null;
+							modifiedBitmap = convertToSepia(flippedImaged);
+							mImageView.setImageBitmap(changeBrightness(
+									modifiedBitmap, brightnessValue));
+							mBitmap = modifiedBitmap;
+
+				    	}catch(OutOfMemoryError o){
+				    		mBitmapResize=getResizedBitmap(mBitmap, (mBitmap.getHeight()/4), (mBitmap.getWidth()/4));
+				    		//mBitmapResize=getResizedBitmap(p[0], (mBitmap.getHeight()/2), (mBitmap.getWidth()/2));
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Remove commented out code.

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 9]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
+							modifiedBitmap=flippedImaged=mBitmapResize;
+
+				    	}
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Again a blank line

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 10]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
                             modifiedBitmap = null;
-                            modifiedBitmap = convertToSepia(flippedImaged);
-                            mImageView.setImageBitmap(changeBrightness(modifiedBitmap, brightnessValue));
+                            modifiedBitmap = convertToAlpha(flippedImaged);
+                            mImageView.setImageBitmap(changeBrightness(
+                                    modifiedBitmap, brightnessValue));
                             mBitmap = modifiedBitmap;
-                        } catch (OutOfMemoryError o) {
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Blank line

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 11]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
+                        }catch(OutOfMemoryError o){
                             mBitmapResize=getResizedBitmap(mBitmap, (mBitmap.getHeight()/4), (mBitmap.getWidth()/4));
                             //mBitmapResize=getResizedBitmap(p[0], (mBitmap.getHeight()/2), (mBitmap.getWidth()/2));
                             modifiedBitmap=flippedImaged=mBitmapResize;
+
                         }
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Blank line

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 12]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
                     }
                 });
+        findViewById(R.id.btnpink).setOnClickListener(
+                new View.OnClickListener() {
+                    public void onClick(View v)
+                    {
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Make this inline

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 13]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
 
-        findViewById(R.id.btnNagative).setOnClickListener(
+                            modifiedBitmap = null;
+                            modifiedBitmap = convertToPink(flippedImaged);
+                            mImageView.setImageBitmap(changeBrightness(
+                                    modifiedBitmap, brightnessValue));
+                            mBitmap = modifiedBitmap;
+
+                        }catch(OutOfMemoryError o){
+                            mBitmapResize=getResizedBitmap(mBitmap, (mBitmap.getHeight()/4), (mBitmap.getWidth()/4));
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Leave a space between RHS or LHS and the equals to assignment.

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 14]
```diff
@@ -589,103 +594,276 @@ public void onClick(View v) {
...
+                    }
+                });
+        findViewById(R.id.btnpolaroid).setOnClickListener(
                 new View.OnClickListener() {
-                    public void onClick(View v) {
-                        try {
-                            // Flags an edit
-                            imageEdited = true;
+                    public void onClick(View v)
+                    {
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Make this inline

Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Make it inline as well

**Comment by madhavrathi** [comment]
@Abhi2424shek Hey i have done all the changes requested.

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 15]
```diff
@@ -88,20 +93,20 @@
...
+    //private static float toDegree = 90;
+    //private static float fromDegree = 0;
     // These are various options can be specified in the intent.
     private Bitmap.CompressFormat mOutputFormat = Bitmap.CompressFormat.JPEG; // only
-                                                                              // used
-                                                                              // with
-                                                                              // mSaveUri
+    // used
+    // with
+    // mSaveUri
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Make this comment in one line

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 16]
```diff
@@ -302,20 +307,20 @@ public void onClick(View v)
 
         Intent intent = getIntent();
         Bundle extras = intent.getExtras();
-       
-        
+
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Two unwanted blank lines

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 17]
```diff
@@ -302,20 +307,20 @@ public void onClick(View v)
...
         if (extras != null) {
             if (extras.getString("circleCrop") != null) {
                 mCircleCrop = true;
                 mAspectX = 1;
                 mAspectY = 1;
             }
-           
-            
+
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Again two blank lines

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 18]
```diff
@@ -436,84 +441,84 @@ public void onClick(View v) {
...
+                if(txtLongtitude.getText().toString().equals("")&& !txtLatitude.getText().toString().equals("")){
+                    AlertDialog.Builder builder=new AlertDialog.Builder(ctx);
+                    builder.setMessage("Please enter Longtitude !");
+                    builder.setPositiveButton("OK", new DialogInterface.OnClickListener()
                     {
                         @Override
                         public void onClick(DialogInterface dialog, int which)
-                        {                        	
+                        {
                         }
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Make this in a single line.

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 19]
```diff
@@ -436,84 +441,84 @@ public void onClick(View v) {
...
+                    builder.create().show();
+                }else if(!txtLongtitude.getText().toString().equals("")&& txtLatitude.getText().toString().equals("")){
+                    AlertDialog.Builder builder=new AlertDialog.Builder(ctx);
+                    builder.setMessage("Please enter Latitude !");
+                    builder.setPositiveButton("OK", new DialogInterface.OnClickListener()
                     {
                         @Override
                         public void onClick(DialogInterface dialog, int which)
-                        {                        	
+                        {
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Make these to single line

Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Here too.

File: app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java [hunk 20]
```diff
@@ -811,12 +927,12 @@ private void onSaveClicked() {
...
                 croppedImage.recycle();
                 croppedImage = b;
-               
+
             }
         }
 
         // Return the cropped image directly or save it to the specified URI.
-        
+
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/gallery3d/media/CropImage.java]: Blank line

**Pull request merged**